### PR TITLE
Pyro

### PR DESCRIPTION
**Pull request by teh**
###### Motivation for this change

Introduce pyro (probabilistic programming language based on pytorch) + one dependency

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/opt-einsum/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  propagatedBuildInputs = [
+    numpy
+  ];
+
+  meta = {
+    description = "Optimizing NumPy's einsum function with order optimization and GPU support.";
+    homepage = http://optimized-einsum.readthedocs.io;
+    license = lib.licenses.mit;
+    platforms = lib.platforms.linux;
```
Comment (FRidh) [inline pkgs/development/python-modules/opt-einsum/default.nix]: Why modify platforms?

File: pkgs/development/python-modules/pyro-ppl/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    python -c "import pyro.distributions"
+    python -c "import pyro.infer"
+    python -c "import pyro.optim"
+  '';
+
+  meta = {
+    description = "A Python library for probabilistic modeling and inference";
+    homepage = http://pyro.ai;
+    license = lib.licenses.mit;
+    platforms = lib.platforms.linux;
```
Comment (FRidh) [inline pkgs/development/python-modules/pyro-ppl/default.nix]: Why modify platforms?

File: pkgs/development/python-modules/opt-einsum/default.nix (hunk 1 above)
Comment (teh) [reply pkgs/development/python-modules/opt-einsum/default.nix]: No reason other than copy-paste. Removed.

File: pkgs/development/python-modules/pyro-ppl/default.nix (hunk 2 above)
Comment (teh) [reply pkgs/development/python-modules/pyro-ppl/default.nix]: In this case it's the same as pytorch but I guess that's implicit in the dependency. Removed.

**Comment by teh** [comment]
@GrahamcOfBorg build python2Packages.pyro-ppl python3Packages.pyro-ppl

**Pull request merged**